cartography/intel/github/repos.py

```diff
@@ -785,6 +929,89 @@ def load_collaborators(
         )
 
 
+@timeit
+def load_python_requirements(
+    neo4j_session: neo4j.Session,
+    update_tag: int,
+    requirements_objects: List[Dict],
+) -> None:
+    query = """
+    UNWIND $Requirements AS req
+        MERGE (lib:PythonLibrary:Dependency{id: req.id})
+        ON CREATE SET lib.firstseen = timestamp(),
+        lib.name = req.name
+        SET lib.lastupdated = $UpdateTag,
+        lib.version = req.version
+
+        WITH lib, req
+        MATCH (repo:GitHubRepository{id: req.repo_url})
+        MERGE (repo)-[r:REQUIRES]->(lib)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = $UpdateTag,
+        r.specifier = req.specifier
+    """
+    neo4j_session.run(
+        query,
+        Requirements=requirements_objects,
+        UpdateTag=update_tag,
+    )
+
+
+@timeit
+def load_github_dependencies(
+    neo4j_session: neo4j.Session,
+    update_tag: int,
+    dependencies: List[Dict],
+) -> None:
+    """
+    Ingest GitHub dependency data into Neo4j using the new data model
+    :param neo4j_session: Neo4J session object for server communication
+    :param update_tag: Timestamp used to determine data freshness
+    :param dependencies: List of dependency objects from GitHub's dependency graph
+    :return: Nothing
+    """
+    # Group dependencies by repository URL for schema-based loading
+    from collections import defaultdict
+
+    dependencies_by_repo = defaultdict(list)
+
+    for dep in dependencies:
+        repo_url = dep["repo_url"]
+        # Remove repo_url from the dependency object since we'll pass it as kwarg
+        dep_without_repo = {k: v for k, v in dep.items() if k != "repo_url"}
+        dependencies_by_repo[repo_url].append(dep_without_repo)
+
+    # Load dependencies for each repository separately
+    for repo_url, repo_dependencies in dependencies_by_repo.items():
+        load_data(
+            neo4j_session,
+            GitHubDependencySchema(),
+            repo_dependencies,
+            lastupdated=update_tag,
+            repo_url=repo_url,
+        )
+
+
+@timeit
+def cleanup_github_dependencies(
+    neo4j_session: neo4j.Session,
+    common_job_parameters: Dict[str, Any],
+    repo_urls: List[str],
+) -> None:
+    """
+    Delete GitHub dependencies and their relationships from the graph if they were not updated in the last sync.
+    :param neo4j_session: Neo4j session
+    :param common_job_parameters: Common job parameters containing UPDATE_TAG
+    :param repo_urls: List of repository URLs to clean up dependencies for
+    """
+    # Run cleanup for each repository separately
+    for repo_url in repo_urls:
```

Comment:
Once the scoped_cleanup stuff is set correctly, you can simplify this by removing the loop and instead just calling

```
    GraphJob.from_node_schema(GitHubDepdencySchema(), common_job_parameters).run(
        neo4j_session
    )
```

this way you don't need to include `repo_url` as a param to the cleanup job anymore.